### PR TITLE
[SW-2372] Expose option used for waiting before the clouding starts in internal backend

### PR DESCRIPTION
**Pull request by jakubhava**
CC: @FengAtH2O

### CODE REVIEW

File: doc/src/site/sphinx/configuration/configuration_properties.rst [hunk 1]
```diff
@@ -264,6 +264,10 @@ Internal backend configuration properties
 |                                                    |                |                                                 | settings and other HDFS-related        |
 |                                                    |                |                                                 | configurations.                        |
 +----------------------------------------------------+----------------+-------------------------------------------------+----------------------------------------+
+| ``spark.ext.h2o.wait.before.clouding``             | ``0``          | ``setWaitBeforeCloudingMillis(Int)``            | ow long we should wait before we start |
```
Comment (mn-mikke) [inline doc/src/site/sphinx/configuration/configuration_properties.rst:267]: typo: `ow long`

**Comment by mn-mikke** [comment]
Any reason for executing the `SpreadRDD`  after expiration of the timeout? Couldn't the SpreadRDD be executed in the loop during the period specified by the timeout value?

**Comment by jakubhava** [comment]
Yup, the reason is that the spread rdd is influenced by options such as num retries, subseq tries etc. During the loops, we might end earlier if these counters hit zero

File: doc/src/site/sphinx/configuration/configuration_properties.rst (hunk 1 above)
Comment (jakubhava) [reply doc/src/site/sphinx/configuration/configuration_properties.rst]: Thanks!

**Comment by jakubhava** [comment]
@mn-mikke What do you think about this change please?

**Comment by jakubhava** [comment]
@mn-mikke thanks for your input! I'll give it another thought

**Comment by jakubhava** [comment]
@mn-mikke I gave it a thought and try to incorporate the timeout into the main build loop. Can you please have a look on the PR one more time?

File: doc/src/site/sphinx/configuration/configuration_properties.rst [hunk 2]
```diff
@@ -264,6 +264,16 @@ Internal backend configuration properties
 |                                                    |                |                                                 | settings and other HDFS-related        |
 |                                                    |                |                                                 | configurations.                        |
 +----------------------------------------------------+----------------+-------------------------------------------------+----------------------------------------+
+| ``spark.ext.h2o.internal.clouding.timeout``        | ``0``          | ``setInternalBackendCloudingTimeout(Int)``      | Specifies how long the clouding should |
```
Comment (mn-mikke) [inline doc/src/site/sphinx/configuration/configuration_properties.rst:267]: `Specifies how long the clouding should last.` -> `Specifies how long the discovering of Spark executors should last.` Clouding up of H2O cluster will happen afterwards, won't it?

File: core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala [hunk 3]
```diff
@@ -115,4 +119,7 @@ object InternalBackendConf {
 
   /** Path to whole Hadoop configuration serialized into XML readable by org.hadoop.Configuration class */
   val PROP_HDFS_CONF: (String, None.type) = ("spark.ext.h2o.hdfs_conf", None)
+
+  /** Timeout for the clouding, unit is milliseconds */
+  val PROP_CLOUDING_TIMEOUT: (String, Int) = ("spark.ext.h2o.internal.clouding.timeout", 0)
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala:124]: WDYT about `spark.ext.h2o.spreadrdd.retries.timeout`?

File: doc/src/site/sphinx/configuration/configuration_properties.rst (hunk 2 above)
Comment (jakubhava) [reply doc/src/site/sphinx/configuration/configuration_properties.rst:267]: Yup, good catch!

File: core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala (hunk 3 above)
Comment (jakubhava) [reply core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala]: 👍

**Pull request merged**